src/nim-light-client/light_client_utils.nim

```diff
@@ -0,0 +1,879 @@
+  FAR_FUTURE_SLOT* = Slot(not 0'u64)
+  FAR_FUTURE_EPOCH* = (not 0'u64).Epoch # 2^64 - 1 in spec
+
+  # FAR_FUTURE_EPOCH* = Epoch(not 0'u64) # in presets
+  FAR_FUTURE_PERIOD* = SyncCommitteePeriod(not 0'u64)
+type
+# Types from base.nim
+  DomainType* = distinct array[4, byte]
+  Eth2Domain* = array[32, byte]
+
+  GraffitiBytes* = distinct array[MAX_GRAFFITI_SIZE, byte]
+  Gwei* = uint64
+  ForkDigest* = distinct array[4, byte]
+
+  # https://github.com/ethereum/consensus-specs/blob/v1.2.0-rc.1/specs/phase0/beacon-chain.md#signedbeaconblockheader
+  SignedBeaconBlockHeader* = object
+    message*: BeaconBlockHeader
+    signature*: ValidatorSig
+
+  # https://github.com/ethereum/consensus-specs/blob/v1.2.0-rc.1/specs/phase0/beacon-chain.md#eth1data
+  Eth1Data* = object
+    deposit_root*: Eth2Digest
+    deposit_count*: uint64
+    block_hash*: Eth2Digest
+
+  # https://github.com/ethereum/consensus-specs/blob/v1.2.0-rc.1/specs/phase0/beacon-chain.md#beaconblockheader
+  BeaconBlockHeader* = object
+    slot*: Slot
+    proposer_index*: uint64 # `ValidatorIndex` after validation
+    parent_root*: Eth2Digest
+    state_root*: Eth2Digest
+    body_root*: Eth2Digest
+
+  # https://github.com/ethereum/consensus-specs/blob/v1.2.0-rc.1/specs/phase0/beacon-chain.md#signingdata
+  SigningData* = object
+    object_root*: Eth2Digest
+    domain*: Eth2Domain
+
+  # https://github.com/ethereum/consensus-specs/blob/v1.2.0-rc.1/specs/phase0/beacon-chain.md#proposerslashing
+  ProposerSlashing* = object
+    signed_header_1*: SignedBeaconBlockHeader
+    signed_header_2*: SignedBeaconBlockHeader
+
+  # https://github.com/ethereum/consensus-specs/blob/v1.2.0-rc.1/specs/phase0/beacon-chain.md#attesterslashing
+  AttesterSlashing* = object
+    attestation_1*: IndexedAttestation
+    attestation_2*: IndexedAttestation
+
+  CommitteeValidatorsBits* = BitList[Limit MAX_VALIDATORS_PER_COMMITTEE]
+
+  # https://github.com/ethereum/consensus-specs/blob/v1.2.0-rc.1/specs/phase0/beacon-chain.md#indexedattestation
+  IndexedAttestation* = object
+    attesting_indices*: List[uint64, Limit MAX_VALIDATORS_PER_COMMITTEE]
+    data*: AttestationData
+    signature*: ValidatorSig
+
+  # https://github.com/ethereum/consensus-specs/blob/v1.2.0-rc.1/specs/phase0/beacon-chain.md#attestation
+  Attestation* = object
+    aggregation_bits*: CommitteeValidatorsBits
+    data*: AttestationData
+    signature*: ValidatorSig
+
+  TrustedAttestation* = object
+    # The Trusted version, at the moment, implies that the cryptographic signature was checked.
+    # It DOES NOT imply that the state transition was verified.
+    # Currently the code MUST verify the state transition as soon as the signature is verified
+    aggregation_bits*: CommitteeValidatorsBits
+    data*: AttestationData
+    signature*: TrustedSig
+
+  # https://github.com/ethereum/consensus-specs/blob/v1.2.0-rc.1/specs/phase0/beacon-chain.md#checkpoint
+  Checkpoint* = object
```

Comment:
Most of the types here are not necessary. Can't they be removed?

Reply:
e97821ffccac0f86933a2a1f5af34c160a1e2ee5